### PR TITLE
Shender/fix project and stage caching

### PR DESCRIPTION
**Pull request by henders**
Cache wasn't taking into account changes to deploy group memberships for stages. So made it part of the cache_key.

project0

==================================================

editing_pods__project0_

/cc @zendesk/runway @zendesk/samson 

### References
 - Jira link:

### Risks
 - None

### CODE REVIEW

File: app/models/stage.rb [hunk 1]
```diff
@@ -191,6 +191,10 @@ def datadog_monitors
     datadog_monitor_ids.to_s.split(/, ?/).map { |id| DatadogMonitor.new(id) }
   end
 
+  def cache_key
+    "#{super}-#{self.deploy_groups.map(&:cache_key)}"
```
Comment (grosser) [inline app/models/stage.rb]: no need for self

**Comment by grosser** [comment]
:+1: a little sad but I guess that's what it takes ...

**Comment by sandlerr** [comment]
Alternatively could we make the update touch the stage, and the stage touch the project?

**Comment by grosser** [comment]
that might be more efficient since we have very few updates

**Comment by henders** [comment]
Phew, was hoping someone would have a better idea, thanks @sandlerr, I'll give that a go.

**Comment by henders** [comment]
@grosser @sandlerr 
Ready for another review!

**Comment by grosser** [comment]
How about having deploy_group_stage.rb to have `belongs_to :stage, touch: true` ?

**Comment by grosser** [comment]
this won't work if you for example update a deploy-group, right ?

**Comment by henders** [comment]
@grosser 
> this won't work if you for example update a deploy-group, right ?

Yea, I figured I'd leave that to another PR if you change the deploygroup name.

> How about having deploy_group_stage.rb to have belongs_to :stage, touch: true ?

Probably even cleaner, I'll give it a go.

**Comment by henders** [comment]
Hmm... nvm, couldn't figure out how to get it working:
```
class DeployGroupsStage < ActiveRecord::Base
  belongs_to :stage, touch: true
  after_save :hello
  def hello
    Rails.logger.info "***************** HEELLLOOOOO "
  end
end
```

So leaving it the way it is.

**Comment by grosser** [comment]
playing with this atm ... stay tuned

**Comment by grosser** [comment]
pushed a fix, works when adding on both sides

**Comment by henders** [comment]
Oh, nice! Thanks @grosser , do you want to +1 your code or do I give the +1 now? :)

**Comment by grosser** [comment]
:+1: But you should +1 and merge too :)

**Comment by henders** [comment]
Happy to do so! :)  :+1:

**Pull request merged**